\begin{center}
Gabe Morris
\end{center}

In [1]:
# Notebook Preamble
import matplotlib.pyplot as plt
import numpy as np
import sympy as sp
from IPython.display import display

plt.style.use('maroon_ipynb.mplstyle')

\pagebreak
\tableofcontents
\pagebreak

\begin{center}
\begin{tabular}{c c c}
ME 8253 & Homework 6 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

#
## Given
A stepped circular rod of 4340 steel (with $S_u$ = 1468 MPa) with diameters of 60 and 45 mm has a root radius of 3 mm at the stepped section. The rod is to be subjected to axial cyclic loading.

![image not found](images/f1.PNG)

The cyclic yield strength ($S^\prime_y$) is estimated from:

$$S^\prime_y=K^\prime(0.002)^{n^\prime}$$

where $K^\prime$ and $n^\prime$ are given in Table A.2

For the purpose of constructing Haigh diagram, exact value of $\sigma_f$ is not needed, as the diagram is not very sensitive to its value. Since $\sigma_f$ is not listed in Table A.2 for this material, we use Eq. 5.20 in the textbook to approximate it as

$$\sigma_f\approx S_u + 345\ (MPa)$$

## Find
Using a Haigh diagram, determine the following for an approximate median fatigue life of $10^6$ cycles:

a. What fully reversed alternating force, $P_a$, can be applied?
b. What is the maximum value of $P_a$, if proper compressive residual stresses are present at the notch root? What is the magnitude of the compressive residual stress needed to obtain this maximum alternating stress?
c. What value of $P_a$ can be applied if the residual stress calculated in (b) is tensile? What fully reversed alternating force, $P_a$, can be applied?

## Solution
According to equation 4.3b, the endurance limit is $700\ MPa$ for materials with an ultimate strength greater than $1400\ MPa$. With the size effect, the endurance limit becomes

$$S_f=0.85(700)=595\ MPa$$

### Part A

![image not found](images/f2.png)

In [2]:
D, d, r = 60, 45, 3  # mm
Su = 1468  # MPa
size_effect = 0.85

if Su <= 1400:
    Sf = 0.5*Su*size_effect
else:
    Sf = 700*size_effect

D/d

1.3333333333333333

In [3]:
r/d

0.06666666666666667

From above the stress concentration factor is $K_t=2$. The fatigue notch factor for the fully reversed condition is

$$K_f=1+\frac{K_t-1}{1+a/r}$$

where $a=0.0254\left(\frac{2070}{S_u}\right)^{1.8}$ with $a$ in mm and $S_u$ in MPa.

In [4]:
Kt = 2
a = 0.0254*(2070/Su)**1.8
a  # mm

0.047149103389883054

In [5]:
Kf = 1 + (Kt - 1)/(1 + a/r)
Kf

1.9845268144780213

Since the stress is maximized at the smaller diameter, 

$$P_a=\frac{S_f}{K_f}\left(\frac{\pi}{4}\right)(d)^2$$

In [6]:
Pa = Sf/Kf*np.pi/4*d**2
Pa  # N

476842.4418454644

### Part B
From Table A.2,

| Property     | Value     |
|--------------|-----------|
| $S_y$        | 1371  MPa |
| $S^\prime_y$ | 863  MPa  |
| $S_f$        | 595   MPa |
| $\sigma_f$   | 1813  MPa |
| $K_f$        | 1.98      |
| $S_{cat}$    | 70    MPa |

$S_{cat}$ comes from the fact that this is a hard steel. The other calculations are shown below.

In [7]:
sig_f = Su + 345
sig_f  # MPa

1813

In [8]:
K_prime, n_prime = 1996, 0.135
Sy_prime = K_prime*0.002**n_prime
Sy_prime  # MPa

862.5804014077875

![image not found](images/f3.png)

In [9]:
Sa = 460  # MPa
Pa = Sa*np.pi/4*d**2
Pa  # N

731598.3892047232

### Part C
From above, point B is when the mean stress is tensile.

In [10]:
Sa = 90
Pa = Sa*np.pi/4*d**2
Pa  # N

143138.81527918496

\pagebreak

\begin{center}
\begin{tabular}{ccc}
ME 8253 & Homework 6 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

#
## Given
An axially loaded member made of 2024-T3 aluminum is repeatedly subjected to the block of stress history shown below.

![image not found](images/f4.PNG)

## Find
a. Complete a summary of the loading block.
b. Using the Basquin equation, $S_{Nf}=\sigma^\prime_f(2N_f)^b$, determine the fatigue strength $S_{Nf}$, the fatigue life $N_f$ and the damage ratio $n/N_f$ for each load segment and estimate the expected life if the member is smooth.
c. Estimate the expected life if the member has a notch with $K_t=2$ and the notch root radius is 1 mm. For the notched member, assume that the given nominal stress block and $K_t$ are based on net stress.

## Solution
### Part A
| Load Segment | $S_{min}$ (MPa) | $S_{max}$ (MPa) | $S_a$ (MPa) | $S_m$ (MPa) | n   |
|--------------|-----------------|-----------------|-------------|-------------|-----|
| 1            | 0               | 300             | 150         | 150         | 100 |
| 2            | -300            | 300             | 300         | 0           | 1   |
| 3            | -300            | 0               | 150         | -150        | 100 |

### Part B
Use the modified Goodman to find $S_{Nf}$ with $S_u=469$ MPa.

$$\frac{S_a}{S_{Nf}}+\frac{S_m}{S_u}=1\rightarrow S_{Nf}=\frac{S_aS_u}{S_u-S_m}$$
$$S_{Nf}=\sigma^\prime_f(2N_f)^b\rightarrow N_f=\frac{1}{2}\left(\frac{S_{Nf}}{\sigma^\prime_f}\right)^{1/b}$$

In [11]:
Su, sig_prime, b = 469, 1100, -0.124

SNf_lamb = lambda Sa___, Su_, Sm_: Sa___*Su_/(Su_ - Sm_)
Nf_lamb = lambda SNf_, sig_prime__, b__: 0.5*(SNf_/sig_prime__)**(1/b__)

# Load 1
SNf1 = SNf_lamb(150, Su, 150)
SNf1  # MPa

220.53291536050156

In [12]:
Nf1 = Nf_lamb(SNf1, sig_prime, b)
Nf1

212496.20843121517

In [13]:
d1 = 100/Nf1
d1

0.0004705966319976477

In [14]:
# Load 2
SNf2 = SNf_lamb(300, Su, 0)
SNf2  # MPa

300.0

In [15]:
Nf2 = Nf_lamb(SNf2, sig_prime, b)
Nf2

17764.216450750755

In [16]:
d2 = 1/Nf2
d2

5.629294164324021e-05

In [17]:
# Load 3
SNf3 = SNf_lamb(150, Su, -150)
SNf3  # MPa

113.65105008077545

In [18]:
Nf3 = Nf_lamb(SNf3, sig_prime, b)
Nf3

44578464.41972726

In [19]:
d3 = 100/Nf3
d3

2.24323563634792e-06

Now the expected life is calculated as the reciprocal of the summation of the damage ratios.

In [20]:
1/sum([d1, d2, d3])  # Blocks

1889.8846990152454

| Load Segment | $S_{Nf}$ (MPa) | $N_f$      | $n$ | $n/N_f$             |
|--------------|----------------|------------|-----|---------------------|
| 1            | 221            | 212496     | 100 | $4.706\cdot10^{-4}$ |
| 2            | 300            | 17764      | 1   | $5.630\cdot10^{-5}$ |
| 3            | 114            | 44,578,464 | 100 | $2.243\cdot10^{-6}$ |
| Total        | -              | -          | -   | 1890                |

### Part C
Everything remains the same, but now the endurance limit is changed to $S_f/K_f$, and a new $b$ is found by assuming that the endurance limit is reached at $10^6$ cycles.

In [21]:
a = 0.0254*(2070/Su)**1.8
a  # mm

0.3676793350247542

In [22]:
Kt, r = 2, 1
Kf = 1 + (Kt - 1)/(1 + a/r)
Kf

1.7311655403361788

In [23]:
Sf = sig_prime*2e6**b
Sf  # MPa

181.9973086280446

In [24]:
Sf_new = Sf/Kf
B = np.log(Sf_new/sig_prime)/np.log(2e6)
B

-0.16182533948270703

The $S_{Nf}$ values stay the same, but the $N_f$ values change.

In [25]:
Nf1 = Nf_lamb(SNf1, sig_prime, B)
Nf1

10274.56557852558

In [26]:
d1 = 100/Nf1
d1

0.009732771593672595

In [27]:
Nf2 = Nf_lamb(SNf2, sig_prime, B)
Nf2

1534.2156409133563

In [28]:
d2 = 1/Nf2
d2

0.0006517988562577001

In [29]:
Nf3 = Nf_lamb(SNf3, sig_prime, B)
Nf3

617791.4135432595

In [30]:
d3 = 100/Nf3
d3

0.00016186693082453747

In [31]:
1/sum([d1, d2, d3])  # Blocks

94.81874910904062

| Load Segment | $S_{Nf}$ (MPa) | $N_f$  | $n$ | $n/N_f$             |
|--------------|----------------|--------|-----|---------------------|
| 1            | 221            | 10275  | 100 | $9.732\cdot10^{-3}$ |
| 2            | 300            | 1534   | 1   | $6.518\cdot10^{-4}$ |
| 3            | 114            | 617791 | 100 | $1.619\cdot10^{-4}$ |
| Total        | -              | -      | -   | 95                  |

\pagebreak

\begin{center}
\begin{tabular}{ccc}
ME 8253 & Homework 6 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

#
## Given
Repeat Problem 2 using the strain-life approach.

## Find
a. Explain why the behavior is linear elastic and the strain amplitude can be calculated using $\epsilon_a=S_a/E$.
b. Use the Smith-Watson-Topper (SWT) equation, $\sigma_{\max } \varepsilon_a E=\left(\sigma_f^{\prime}\right)^2\left(2 N_f\right)^{2 b}+\sigma_f^{\prime} \varepsilon_f^{\prime} E\left(2 N_f\right)^{b+c}$, to account for the mean stress effect, and the fatigue properties from Table A.2 to determine the fatigue life and calculate the damage ratio for each load segment.
c. Estimate the total fatigue life in terms of blocks.

For a notched member with $K_t=2$ and a root radius of 1 mm:

d. Explain why the behavior is inelastic.
e. Determine the fatigue notch factor $K_f$.
f. Determine the stress range $S$ from the beginning of the block (Point O) to point A. Using Neuber's rule, $\frac{\sigma_A^2}{E}+\sigma_A\left(\frac{\sigma_A}{K}\right)^{1 / n}=\varepsilon_A \sigma_A=\frac{\left(K_f S_A\right)^2}{E}$, with the cyclic stress-strain equation, $\varepsilon_a=\frac{\sigma_a}{E}+\left(\frac{\sigma_a}{K^{\prime}}\right)^{1 / n^{\prime}}$, find the notch stress $\sigma_A$ and the strain $\epsilon_A$ at point A.
g. For load segment 1 (point A to point B), determine the stress range $\Delta S$ from point A to point B. Using the Neuber's rule, $\frac{(\Delta \sigma)^2}{E}+2 \Delta \sigma\left(\frac{\Delta \sigma}{2 K^{\prime}}\right)^{\frac{1}{n^{\prime}}}=\frac{\left(K_f \Delta S\right)^2}{E}$, with the cyclic stress equation, $\Delta \varepsilon=\frac{\Delta \sigma}{2 E}+\left(\frac{\Delta \sigma}{2 K^{\prime}}\right)^{\frac{1}{n^{\prime}}}$, find the notch stress $\sigma_B=\sigma_A-\Delta\sigma$ and strain $\epsilon_B=\epsilon_A-\Delta\epsilon$ at point B. Determine the strain amplitude $\epsilon_a$ and the maximum stress $\sigma_{max}$ and calculate the fatigue life $N_f$ using the SWT equation.
h. Repeat (g) for load segment 2 (point C to point D).
i. Repeat (g) for load segment 3 (point D to point E).
j. Estimate the fatigue life in terms of blocks.

## Solution
Recall the following,

| Load Segment | $S_{min}$ (MPa) | $S_{max}$ (MPa) | $S_a$ (MPa) | $S_m$ (MPa) | n   |
|--------------|-----------------|-----------------|-------------|-------------|-----|
| 1            | 0               | 300             | 150         | 150         | 100 |
| 2            | -300            | 300             | 300         | 0           | 1   |
| 3            | -300            | 0               | 150         | -150        | 100 |

### Part A
The behavior is mostly elastic because the stress amplitude is 150 MPa for most of the loading duration. Since the stress amplitude is less than half of the material's yield strength ($S_y=379$ MPa), the duration is of the load is mostly elastic. Also, even the peak stress values (300 MPa) is less than the yield strength.

### Part B
The below code cell is a contains a function that will numerically solve for $N_f$ using the Smith-Watson-Topper equation.

In [32]:
# Define properties
E, sig_prime = 70_000, 1100  # MPa
eps_prime, b, c = sp.S('0.22'), sp.S('-0.124'), sp.S('-0.59')

# Define symbols
sig_max_, eps_a_, E_, sig_prime_, Nf_, eps_prime_, b_, c_= sp.symbols(r"\sigma_{max} \epsilon_a E \sigma^'_f N_f \epsilon^'_f b c")

def SWT_elastic(S_max, eps_a, Nf_guess=10_000):
    sub_list = [(sig_max_, S_max), (eps_a_, eps_a), (E_, E), (sig_prime_, sig_prime), (eps_prime_, eps_prime), (b_, b), (c_, c)]
    eq = sp.Eq(sig_max_*eps_a_*E_, sig_prime_**2*(2*Nf_)**(2*b_) + sig_prime_*eps_prime_*E_*(2*Nf_)**(b_ + c_))
    display(eq)

    with sp.evaluate(False):
        eq_sub = sp.Eq(eq.lhs.subs(sub_list), eq.rhs.subs(sub_list))
        display(eq_sub)

    try:
        sol = sp.nsolve(eq_sub, Nf_guess)
    except ValueError:
        sol = sp.oo
    display(sp.Eq(Nf_, sol))
    return sol

# Load Segment 1
S_max1, eps_a1 = 300, sp.S('0.0021429')
Nf1 = SWT_elastic(S_max1, eps_a1)
d1 = 100/Nf1

Eq(E*\epsilon_a*\sigma_{max}, E*\epsilon^'_f*\sigma^'_f*(2*N_f)**(b + c) + \sigma^'_f**2*(2*N_f)**(2*b))

Eq(70000*0.0021429*300, 1100**2*(2*N_f)**(2*(-0.124)) + 70000*0.22*1100*(2*N_f)**(-0.59 - 0.124))

Eq(N_f, 324108.948236852)

In [33]:
# Load Segment 2
S_max2, eps_a2 = 300, 2*eps_a1
Nf2 = SWT_elastic(S_max2, eps_a2)
d2 = 1/Nf2

Eq(E*\epsilon_a*\sigma_{max}, E*\epsilon^'_f*\sigma^'_f*(2*N_f)**(b + c) + \sigma^'_f**2*(2*N_f)**(2*b))

Eq(70000*0.0042858*300, 1100**2*(2*N_f)**(2*(-0.124)) + 70000*0.22*1100*(2*N_f)**(-0.59 - 0.124))

Eq(N_f, 25160.2548136955)

In [34]:
S_max3, eps_a3 = 0, eps_a1
Nf3 = SWT_elastic(S_max3, eps_a3)
d3 = 100/Nf3

Eq(E*\epsilon_a*\sigma_{max}, E*\epsilon^'_f*\sigma^'_f*(2*N_f)**(b + c) + \sigma^'_f**2*(2*N_f)**(2*b))

Eq(70000*0.0021429*0, 1100**2*(2*N_f)**(2*(-0.124)) + 70000*0.22*1100*(2*N_f)**(-0.59 - 0.124))

Eq(N_f, oo)

Here is a table of results.

| Load Segment | $S_a$ (MPa) | $\epsilon_a$ | $\sigma_{max}$ (MPa) | $\sigma_{max}\epsilon_a$ | $n$ | $N_f$    | $n/N_f$            |
|--------------|-------------|--------------|----------------------|--------------------------|-----|----------|--------------------|
| 1            | 150         | 0.0021429    | 300                  | 0.64287                  | 100 | 324109   | $3.09\cdot10^{-4}$ |
| 2            | 300         | 0.0042858    | 300                  | 1.28571                  | 1   | 25160    | $3.97\cdot10^{-5}$ |
| 3            | 150         | 0.0021429    | 0                    | 0                        | 100 | $\infty$ | 0                  |
| Total        | -           | -            | -                    | -                        | -   | -        | 2871               |

### Part C

In [35]:
print([d1, d2, d3])
1/sum([d1, d2, d3])  # Blocks

[0.000308538226247682, 3.97452254519963e-5, 0]


2871.22455896151

### Part D
For the notched case with a stress concentration factor of $K_t=2$, the peak stresses reach values of $2\cdot300=600$ MPa. This exceeds the yield strength value of 379 MPa, so the behavior is in the plastic region of the stress-strain curve.

### Part E
The fatigue notch factor is the same as the previous problem since the stress concentration factor and root radius is the same.

In [36]:
a, Kf

(0.3676793350247542, 1.7311655403361788)

### Part F
The below code cell contains a function that will solve for $\sigma_A$ and $\epsilon_A$.

In [37]:
# Define more properties
n, K = sp.S('0.032'), 455
n_prime, K_prime = sp.S('0.065'), 655

# Define more symbols
sig_a_, eps_A_, K_, n_, Sa_, Kf_, K_prime_, n_prime_ = sp.symbols(r"\sigma_A \epsilon_A K n S_A K_f K^' n^'")

def Neuber_f(Sa__):
    sub_list = [(E_, E), (K_, K), (n_, n), (Kf_, Kf), (Sa_, Sa__), (K_prime_, K_prime), (n_prime_, n_prime)]

    eq1 = sp.Eq(sig_a_**2/E_ + sig_a_*(sig_a_/K_)**(1/n_), (Kf_*Sa_)**2/E_)
    eq2 = sp.Eq(eps_A_, sig_a_/E_ + (sig_a_/K_prime_)**(1/n_prime_))
    display(eq1, eq2)

    with sp.evaluate(False):
        eq1_new = sp.Eq(eq1.lhs.subs(sub_list), eq1.rhs.subs(sub_list))
        eq2_new = sp.Eq(eq2.lhs.subs(sub_list), eq2.rhs.subs(sub_list))
        display(eq1_new, eq2_new)

    sol = sp.nsolve([eq1_new, eq2_new], (sig_a_, eps_A_), [50, 0.03], dict=True)[0]
    for key, value in sol.items():
        display(sp.Eq(key, value))

    return sol

f = Neuber_f(300)

Eq(\sigma_A*(\sigma_A/K)**(1/n) + \sigma_A**2/E, K_f**2*S_A**2/E)

Eq(\epsilon_A, (\sigma_A/K^')**(1/n^') + \sigma_A/E)

Eq(\sigma_A**2/70000 + \sigma_A*(\sigma_A/455**1)**(1/0.032), 1.73116554033618**2*300**2/70000**1)

Eq(\epsilon_A, \sigma_A/70000**1 + (\sigma_A/655**1)**(1/0.065))

Eq(\sigma_A, 382.990491552314)

Eq(\epsilon_A, 0.00573104808066133)

### Part G
The below code cell contains a function that will solve for $\Delta\sigma$ and $\Delta\epsilon$.

In [38]:
# Define more symbols
del_sig_, del_eps_ = sp.symbols(r'\Delta\sigma \Delta\epsilon')
del_S_ = sp.Symbol(r'\Delta S')

def Neuber(del_S):
    sub_list = [(E_, E), (Kf_, Kf), (del_S_, del_S), (K_prime_, K_prime), (n_prime_, n_prime)]

    eq1 = sp.Eq(del_sig_**2/E_ + 2*del_sig_*(del_sig_/(2*K_prime_))**(1/n_prime_), (Kf_*del_S_)**2/E_)
    eq2 = sp.Eq(del_eps_, del_sig_/(2*E_) + (del_sig_/(2*K_prime_))**(1/n_prime_))
    display(eq1, eq2)

    with sp.evaluate(False):
        eq1_new = sp.Eq(eq1.lhs.subs(sub_list), eq1.rhs.subs(sub_list))
        eq2_new = sp.Eq(eq2.lhs.subs(sub_list), eq2.rhs.subs(sub_list))
        display(eq1_new, eq2_new)

    sol = sp.nsolve([eq1_new, eq2_new], (del_sig_, del_eps_), [50, 0.02], dict=True)[0]
    for key, value in sol.items():
        display(sp.Eq(key, value))

    return sol

g = Neuber(300)

Eq(2*\Delta\sigma*(\Delta\sigma/(2*K^'))**(1/n^') + \Delta\sigma**2/E, K_f**2*\Delta S**2/E)

Eq(\Delta\epsilon, (\Delta\sigma/(2*K^'))**(1/n^') + \Delta\sigma/(2*E))

Eq(\Delta\sigma**2/70000 + 2*\Delta\sigma*((1/2)*\Delta\sigma/655**1)**(1/0.065), 1.73116554033618**2*300**2/70000**1)

Eq(\Delta\epsilon, (1/2)*\Delta\sigma/70000**1 + ((1/2)*\Delta\sigma/655**1)**(1/0.065))

Eq(\Delta\sigma, 519.303660550288)

Eq(\Delta\epsilon, 0.00370996905518844)

The notch stress and strain at point B is calculated below.

In [39]:
sig_B = f[sig_a_] - g[del_sig_]
sig_B  # MPa

-136.313168997974

In [40]:
eps_B = f[eps_A_] - g[del_eps_]
eps_B

0.00202107902547290

As for the fatigue life for load segment 1, it can be found by using the SWT with the maximum stress ($\sigma_{max}$) being the maximum notch stress at point A. The strain amplitude is

$$\epsilon_a=\frac{\epsilon_A-\epsilon_B}{2}$$

In [41]:
eps_a1 = (f[eps_A_] - eps_B)/2
eps_a1

0.00185498452759422

In [42]:
Nf1 = SWT_elastic(f[sig_a_], eps_a1)

Eq(E*\epsilon_a*\sigma_{max}, E*\epsilon^'_f*\sigma^'_f*(2*N_f)**(b + c) + \sigma^'_f**2*(2*N_f)**(2*b))

Eq(70000*0.00185498452759422*382.990491552314, 1100**2*(2*N_f)**(2*(-0.124)) + 70000*0.22*1100*(2*N_f)**(-0.59 - 0.124))

Eq(N_f, 221182.655322801)

In [43]:
d1 = 100/Nf1
d1

0.000452115017129425

### Part H
The stress range for point C to point D is 300 to -300; therefore, $\Delta S=600$ MPa.

In [44]:
h = Neuber(600)

Eq(2*\Delta\sigma*(\Delta\sigma/(2*K^'))**(1/n^') + \Delta\sigma**2/E, K_f**2*\Delta S**2/E)

Eq(\Delta\epsilon, (\Delta\sigma/(2*K^'))**(1/n^') + \Delta\sigma/(2*E))

Eq(\Delta\sigma**2/70000 + 2*\Delta\sigma*((1/2)*\Delta\sigma/655**1)**(1/0.065), 1.73116554033618**2*600**2/70000**1)

Eq(\Delta\epsilon, (1/2)*\Delta\sigma/70000**1 + ((1/2)*\Delta\sigma/655**1)**(1/0.065))

Eq(\Delta\sigma, 884.827904134403)

Eq(\Delta\epsilon, 0.00870949255504041)

In [45]:
sig_D = f[sig_a_] - h[del_sig_]
sig_D  # MPa

-501.837412582090

In [46]:
eps_D = f[eps_A_] - h[del_eps_]
eps_D

-0.00297844447437907

The strain amplitude for load segment 2 is,

$$\epsilon_a=\frac{\epsilon_C-\epsilon_D}{2}$$

The maximum stress is the notch stress at point C ($\sigma_{max}=\sigma_C=\sigma_A=383$ MPa).

In [47]:
eps_a2 = (f[eps_A_] - eps_D)/2
eps_a2

0.00435474627752020

In [48]:
Nf2 = SWT_elastic(f[sig_a_], eps_a2)

Eq(E*\epsilon_a*\sigma_{max}, E*\epsilon^'_f*\sigma^'_f*(2*N_f)**(b + c) + \sigma^'_f**2*(2*N_f)**(2*b))

Eq(70000*0.0043547462775202*382.990491552314, 1100**2*(2*N_f)**(2*(-0.124)) + 70000*0.22*1100*(2*N_f)**(-0.59 - 0.124))

Eq(N_f, 10406.2767567)

In [49]:
d2 = 1/Nf2
d2

9.60958490130638e-5

### Part I
The stress range from point D to point E is -300 MPa, which should result in the same notch stress/strain ranges from load segment 1 because the stress range is independent of the sign/direction.

In [50]:
i = Neuber(-300)

Eq(2*\Delta\sigma*(\Delta\sigma/(2*K^'))**(1/n^') + \Delta\sigma**2/E, K_f**2*\Delta S**2/E)

Eq(\Delta\epsilon, (\Delta\sigma/(2*K^'))**(1/n^') + \Delta\sigma/(2*E))

Eq(\Delta\sigma**2/70000 + 2*\Delta\sigma*((1/2)*\Delta\sigma/655**1)**(1/0.065), 1.73116554033618**2*(-300)**2/70000**1)

Eq(\Delta\epsilon, (1/2)*\Delta\sigma/70000**1 + ((1/2)*\Delta\sigma/655**1)**(1/0.065))

Eq(\Delta\sigma, 519.303660550288)

Eq(\Delta\epsilon, 0.00370996905518844)

In [51]:
sig_E = sig_D + i[del_sig_]
sig_E  # MPa

17.4662479681981

In [52]:
eps_E = eps_D + i[del_eps_]
eps_E

0.000731524580809363

The maximum stress is the max between the notch stress $\sigma_D$ and $\sigma_E$; therefore, $\sigma_{max}=17.5$ MPa. The strain amplitude is,

$$\epsilon_a=\frac{\epsilon_E-\epsilon_D}{2}$$

In [53]:
eps_a3 = (eps_E - eps_D)/2
eps_a3

0.00185498452759422

In [54]:
Nf3 = SWT_elastic(sig_E, eps_a3)

Eq(E*\epsilon_a*\sigma_{max}, E*\epsilon^'_f*\sigma^'_f*(2*N_f)**(b + c) + \sigma^'_f**2*(2*N_f)**(2*b))

Eq(70000*0.00185498452759422*17.4662479681981, 1100**2*(2*N_f)**(2*(-0.124)) + 70000*0.22*1100*(2*N_f)**(-0.59 - 0.124))

Eq(N_f, 49626794988.1148)

In [55]:
d3 = 100/Nf3
d3

2.01504046400637e-9

The final table of results is,

| Load Segment | $\Delta_S$ (MPa) | $\epsilon_a$ | $\sigma_{max}$ (MPa) | $\sigma_{max}\epsilon_a$ | $n$ | $N_f$    | $n/N_f$            |
|--------------|------------------|--------------|----------------------|--------------------------|-----|----------|--------------------|
| 1            | 300              | 0.00185      | 383                  | 0.71                     | 100 | 221182   | $3.09\cdot10^{-4}$ |
| 2            | 600              | 0.00435      | 383                  | 1.7                      | 1   | 10406    | $3.97\cdot10^{-5}$ |
| 3            | -300             | 0.00185      | 17.5                 | 0.03                     | 100 | $\infty$ | 0                  |
| Total        | -                | -            | -                    | -                        | -   | -        | 1824               |


In [56]:
1/sum([d1, d2, d3])

1824.10890791578